float64 mass
float64 offset
float64 angle
